### ISSUE TITLE
Crash with CALayerInvalidGeometry uncaught exception

### ISSUE THREAD

**Issue by RRybakowski** [teads/TeadsSDK-iOS#60]
I was testing your demo app, and when i tried opening eg. the "inRead Top ScrollView" example, i experienced a crash. It happens only when i very quickly dismiss the view controller that is loading the ad. This is what i get in the console (no useful stack trace)

```
2016-09-21 10:19:25.296 TeadsSDKDemo-cal[3414:2443720] *** Terminating app due to uncaught exception 'CALayerInvalidGeometry', reason: 'CALayer position contains NaN: [nan nan]'
*** First throw call stack:
(0x180f25900 0x180593f80 0x180f25848 0x183619800 0x18361975c 0x1836196f4 0x185c0fbe8 0x10011ed0c 0x1000ee2a0 0x1000eeb5c 0x1001192b8 0x100106450 0x100124274 0x1002fdbf0 0x1002fdbb0 0x100303658 0x180edcbb0 0x180edaa18 0x180e09680 0x182318088 0x185c80d90 0x1000df6dc 0x1809aa8b8)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

This also happened when i added your SDK to my project, again when quickly closing the view controller loading the ad.

**Comment by Niko-r** [comment]
Hi @RRybakowski, we'll look into this.

Thanks for reporting

**Comment by Niko-r** [comment]
Fixed with last release v2.4.2
